Add UserDropdownDisclosure

ashleemboyer opened the pull request:
## 🔗 Relevant links

<!--
Include links to the branch preview, Asana task, and Figma designs wherever possible to make reviewing your PR easier.
When including the preview link, make sure to remove any '.' characters from the branch name:
  - Example: ks.my-branch -> ksmy-branch
-->

- [Preview link](https://dev-portal-git-ambadd-user-dropdown-hashicorp.vercel.app/swingset/components/userdropdowndisclosure) 🔎 
- [Asana task](https://app.asana.com/0/1202097197789424/1202665629707449/f) 🎟️

## 🗒️ What

<!--
Briefly list out the changes proposed in this PR.
-->

- Adds new `UserDropdownDisclosure` component + Swingset page

## 🤷 Why

<!--
Describe why the changes proposed are needed. Some examples: new feature requested, refactor to make things easier later, styling tweaks requested by design, etc.
-->

- In preparation for: [[NavigationHeader] Render UserDropdown (behind enable_auth flag) on wide viewports when a user is authenticated](https://app.asana.com/0/1202520168081556/1202665629707458/f)

## 🛠️ How

<!--
Dive into the approach you took, list resources you referenced, detail other approaches you tried but didn't end up going with, etc.
-->

- This component is built on top of `DropdownDisclosure`

## 📸 Design Screenshots

<!--
Include a screenshot or two and a link to the designs you referenced with this code. These are both helpful context for reviewers to understand what designs you were looking at when you were putting this code together.
-->

[Figma link](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=11303%3A108294)

<img width="600" alt="" src="https://user-images.githubusercontent.com/43934258/181582290-84626427-5dfb-4f38-a178-3f874f6b9dd5.png">

## 🧪 Testing

<!--
Create a checklist for going through how to test your proposed changes. If there is anything to configure before interacting with the project in a browser, such as toggling feature flags, changing machine settings, or simulating behavior in browser dev tools, list those steps first.

- [ ] Step 1
- [ ] Step 2
- [ ] Step 3
- [ ] ...
-->

- [ ] Go to [the new Swingset page](https://dev-portal-git-ambadd-user-dropdown-hashicorp.vercel.app/swingset/components/userdropdowndisclosure)
- [ ] Make sure the examples work as you expect
- [ ] Make sure the playground works as you expect

ashleemboyer commented on src/components/toast/index.tsx:
```diff
@@ -90,7 +90,8 @@ function toast({
 const developmentToast = (...args: Parameters<typeof toast>) => {
 	if (process.env.NODE_ENV !== 'production') {
-		return toast(...args)
+		const [options, ...restArgs] = args
+		return toast({ ...options, autoDismiss: false }, ...restArgs)
```
🗒️ These changes make all `developementToast`s _not_ `autoDismiss`.

kendallstrautman commented on src/components/user-dropdown-disclosure/docs.mdx:
```diff
@@ -0,0 +1,126 @@
+
+<UserDropdownDisclosure
+	user={{
+		image: 'https://placekitten.com/g/100/100',
```
😻

kendallstrautman reviewed (approved):
Looks great! A few comments but nothing blocking. 

Q: should we handle a max width for the dropdown panel in case a users email is super long? I could imagine this causing the panel to bleed off the page. 
<img width="478" alt="Screen Shot 2022-07-28 at 9 54 57 AM" src="https://user-images.githubusercontent.com/36613477/181595065-55ec8434-e550-4d67-adb6-7cc676ae38ac.png">

Another Q: Does the dropdown panel 'know' when to float to the left below the button? Just looking at the in-situ designs, all of the panels float left. I'm not sure if that logic was handled in another PR.

ashleemboyer commented:
> Q: should we handle a max width for the dropdown panel in case a users email is super long? I could imagine this causing the panel to bleed off the page.

Ooooh good catch. That's more in the scope of `DropdownDisclosure`, so I'm going to go see what the HDS component does and will update in a follow-up PR! 😊

> Another Q: Does the dropdown panel 'know' when to float to the left below the button? Just looking at the in-situ designs, all of the panels float left. I'm not sure if that logic was handled in another PR.

Also good catch! I meant to expose a `listPosition` prop as well, since `DropdownDisclosure` exposes that. I've updated this and added an example in https://github.com/hashicorp/dev-portal/pull/744/commits/a2ef6049293e67226c00bfbdb55d929f10c3653b. 😁
